src/inpod/workloadmanager.rs

```diff
@@ -0,0 +1,586 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::readiness;
+use drain::Watch;
+use std::path::PathBuf;
+use std::time::Duration;
+use tokio::net::UnixStream;
+use tracing::{debug, error, info, warn};
+
+use super::statemanager::WorkloadProxyManagerState;
+use super::Error;
+
+#[mockall_double::double]
+use super::protocol::WorkloadStreamProcessor;
+
+const RETRY_DURATION: Duration = Duration::from_secs(5);
+
+struct WorkloadProxyNetworkHandler {
+    uds: PathBuf,
+}
+
+struct WorkloadProxyReadinessHandler {
+    ready: readiness::Ready,
+    // Manually drop as we don't want to mark ready if we are dropped.
+    // This can happen when the server drains.
+    block_ready: Option<std::mem::ManuallyDrop<readiness::BlockReady>>,
+}
+
+pub struct WorkloadProxyManager {
+    state: super::statemanager::WorkloadProxyManagerState,
+    networking: WorkloadProxyNetworkHandler,
+    // readiness - we are only ready when we are connected. if we get disconnected, we become not ready.
```

Comment:
what is the meaning/use of transitioning from ready -> not ready?

Currently its only not ready -> ready.

Reply:
My thinking was that the user wants to be aware everytime the ztunnel is not connected to the CNI agent (as new pods will not get traffic).
The way to signal that is to make the ztunnel not-ready when it looses connectivity to the node agent.

so the meaning of "Ready" here is that ztunnel is ready to operate on new pods (i.e. connected to node agent).
WDYT?